pkg/types/aws/platform.go

```diff
@@ -33,8 +34,15 @@ type Platform struct {
 	// resources will be created.  Leave unset to have the installer
 	// create subnets in a new VPC on your behalf.
 	//
+	// Deprecated: use platform.aws.vpc.subnets
+	//
+	// +optional
+	DeprecatedSubnets []string `json:"subnets,omitempty"`
+
+	// VPC specifies the VPC configuration for the cluster.
+	//
 	// +optional
-	Subnets []string `json:"subnets,omitempty"`
+	VPC `json:"vpc,omitempty"`
```

Comment:
nit: one side effect of not using pointer here is that the default generated `install-config.yaml` for AWS will have an extra empty VPC field when running `openshift-install create install-config --dir=<dir>`

```yaml
  aws:
    region: us-east-1
    vpc: {}
```

Is this okay?

Reply:
hmm, I guess it will introduce quite a refactor mess when using pointer (i.e. have to check nil). I will keep it this way).